modules/nextflow/src/main/groovy/nextflow/executor/PbsExecutor.groovy

```diff
@@ -49,8 +49,19 @@ class PbsExecutor extends AbstractGridExecutor {
             result << '-q'  << (String)task.config.queue
         }
 
+        // cluster options
+        if( task.config.clusterOptions ) {
+            result << task.config.clusterOptions.toString() << ''
+        }
+
+        // task cpus
         if( task.config.cpus > 1 ) {
-            result << '-l' << "nodes=1:ppn=${task.config.cpus}"
+            if( task.config.clusterOptions && task.config.clusterOptions.indexOf('-l') != -1 ) {
```

Comment:
would not be better to check specifically for `-l nodes=` and `-l mem=` instead of genetic `-l` option?

Reply:
I don't think so

Reply:
If the clusterOptions specifies `-l walltime=something` why it should conflict with `cpus` directive?

Reply:
In that case you can use the `time` directive instead

Reply:
From the online manual it seems that the occurence of `nodes/ mem` etc is not ordered therefore relying upon the baseline `-l` seems like a good solution.


Reference: https://docs.adaptivecomputing.com/torque/4-0-2/Content/topics/commands/qsub.htm

<img width="926" alt="image" src="https://user-images.githubusercontent.com/12799326/204624014-4563d0b6-a787-4065-ba17-29b49c1077f9.png">

Reply:
Therefore the clusterOptions string should be tokenised by `,` and check each option punctually.

Reply:
why?

Reply:
In case there are counterparts to `-l` for other HPC executors then perhaps it makes sense to implement a general logic for tokenizing across executors but in case it's only PBS specific then perhaps the override makes sense?

Reply:
As far as I can tell, other HPC executors don't have this problem, because they take resource arguments one at a time.

Reply:
I agreed with @bentsherman, the problem is that pbs `-l` can take a myriad of options, from queue to architecture, and even custom node proprieties combined with `ppn` which specifies cpus

Reply:
@pditommaso, shall we take this forward?